sdk/core/azure-core/src/test/java/com/azure/core/util/ContextTests.java

```diff
@@ -111,4 +112,18 @@ public void getValues(Context context, Map<Object, Object> expected) {
             Arguments.of(contextWithMultipleSameKeys, Collections.singletonMap("key", "value2"))
         );
     }
+
+    @Test
+    public void getValueNullParent() {
```

Comment:
Do we have JMH benchmark for measuring the perf improvements with this change?

Reply:
No, we don't as yet. Once the PR is functionally correct and passing unit tests, we should definitely create benchmarks and validate that this is a win, because if it isn't we should bin it due to increased complexity of code.